src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs

```diff
@@ -2,16 +2,63 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+#if (GenerateApiOrGraph)
+using Microsoft.Extensions.Configuration;
+using Microsoft.Identity.Web;
+using System.Net;
+using System.Net.Http;
+#endif
+#if (GenerateGraph)
+using Microsoft.Graph;
+#endif
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.Extensions.Logging;
 
 namespace Company.WebApplication1.Pages
 {
+#if (GenerateApiOrGraph)
+    [AuthorizeForScopes(ScopeKeySection = "CalledApi:CalledApiScopes")]
+#endif
     public class IndexModel : PageModel
     {
         private readonly ILogger<IndexModel> _logger;
 
+#if (GenerateApi)
+        private readonly IDownstreamWebApi _downstreamWebApi;
+
+        public IndexModel(ILogger<IndexModel> logger,
+                          IDownstreamWebApi downstreamWebApi)
+        {
+             _logger = logger;
+            _downstreamWebApi = downstreamWebApi;
+       }
+
+        public async Task OnGet()
+        {
+            ViewData["ApiResult"] = await _downstreamWebApi.CallWebApi();
+
+            // You can also specify the relative endpoint and the scopes
+            // ViewData["ApiResult"] = await _downstreamWebApi.CallWebApi("me",
+            //                                                             new string[] {"user.read"});
```

Comment:
I'm a bit hesitant to litter the template with comments like this one. I would rather have a comment instead with a link to an official doc explaining the additional scenarios in detail. Users are going to notice this the moment they modify .CallWebApi above (intellisense will offer the overload).

Reply:
fair enough @javiercn